src/Core/Domain/Product/SpecificPrice/QueryResult/SpecificPriceForListing.php

```diff
@@ -0,0 +1,265 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\QueryResult;
+
+use DateTimeInterface;
+use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\ValueObject\Price;
+
+class SpecificPriceForListing
+{
+    /**
+     * @var int
+     */
+    private $specificPriceId;
+
+    /**
+     * @var string
+     */
+    private $reductionType;
+
+    /**
+     * @var DecimalNumber
+     */
+    private $reductionValue;
+
+    /**
+     * @var bool
+     */
+    private $includesTax;
+
+    /**
+     * @var Price
+     */
+    private $price;
+
+    /**
+     * @var int
+     */
+    private $fromQuantity;
+
+    /**
+     * @var string
+     */
+    private $shop;
+
+    /**
+     * @var string
+     */
+    private $currency;
+
+    /**
+     * @var string
+     */
+    private $country;
+
+    /**
+     * @var string
+     */
+    private $group;
+
+    /**
+     * @var string
+     */
+    private $customer;
+
+    /**
+     * @var DateTimeInterface
+     */
+    private $dateTimeFrom;
+
+    /**
+     * @var DateTimeInterface
+     */
+    private $dateTimeTo;
+
+    /**
+     * @var string|null
+     */
+    private $combinationName;
+
+    /**
+     * @param int $specificPriceId
+     * @param string $reductionType
+     * @param DecimalNumber $reductionValue
+     * @param bool $includesTax
+     * @param Price $price
+     * @param int $fromQuantity
+     * @param DateTimeInterface $dateTimeFrom
+     * @param DateTimeInterface $dateTimeTo
+     * @param string|null $combinationName
+     * @param string|null $shop
+     * @param string|null $currency
+     * @param string|null $country
+     * @param string|null $group
+     * @param string|null $customer
+     */
+    public function __construct(
+        int $specificPriceId,
+        string $reductionType,
+        DecimalNumber $reductionValue,
+        bool $includesTax,
+        Price $price,
+        int $fromQuantity,
+        DateTimeInterface $dateTimeFrom,
+        DateTimeInterface $dateTimeTo,
+        ?string $combinationName,
+        ?string $shop,
+        ?string $currency,
+        ?string $country,
+        ?string $group,
+        ?string $customer
+    ) {
+        $this->specificPriceId = $specificPriceId;
+        $this->reductionType = $reductionType;
```

Comment:
As I saw in many QueryResults we are not using value objects. (I guess the reason was too avoid validating them redundantly because they come from database and we are using them only for reading). I don't know if that was right though, but it made me stick to scalar values here.